plugins/widget/plugin.js

```diff
@@ -3245,10 +3282,33 @@
 
 			// Prevent cutting in read-only editor (#1570).
 			if ( isCut && !editor.readOnly ) {
-				widget.repository.del( widget );
-				editor.fire( 'saveSnapshot' );
+				handleCut();
 			}
 		}, 100 ); // Use 100ms, so Chrome (@Mac) will be able to grab the content.
+
+		function getClipboardHtml() {
+			var selectedHtml = editor.getSelectedHtml( true );
+
+			if ( editor.widgets.focused ) {
+				return editor.widgets.focused.getClipboardHtml();
```

Comment:
I'm wondering if there is still need for a single widget copy handling :thinking: Maybe we could simplify it into a single code path by operating only on selection?

Reply:
Unfortunately something's wrong with selection then 😞